### PR TITLE
Use bolt for database security

### PR DESCRIPTION
**Pull request by DadImScared**

### CODE REVIEW

File: rules.bolt [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+
+type WriteOnlyCoordinator {
+  write() { hasRole("coordinator") }
+}
+
+type OnlyNew {
+  validate() { prior(this) == null }
+}
+
+type EditStatus extends WriteOnlyCoordinator {
```
Comment (anantakrishna) [inline rules.bolt]: I think we can avoid this for now, as various phases will have different statuses. Remove this please.

File: rules.bolt [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+      }
+    }
+  }
+}
+
+/original is ReadOnlyCoordinator {
+  /{list} {
+    index() { ["soundQualityReporting/status"] }
+    /{fileName} {
+      /contentReporting {
```
Comment (anantakrishna) [inline rules.bolt]: This whole block repeats below as well. And it is going to be the same for track editing also. Can we extract some common block?

File: rules.bolt [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+        )
+      }
+      /complete is OnlyNew {}
+      /created is OnlyNew {}
+    }
+  }
+}
+
+/edited/{list}/{fileName} is ReadOnlyCoordinator {
+  /trackEditing {
```
Comment (anantakrishna) [inline rules.bolt]: Here again two blocks are identical. Better to refactor and use common definition.

File: firebase.json [hunk 4]
```diff
@@ -1,6 +1,6 @@
 {
   "database": {
-    "rules": "./database.rules.json"
+    "rules": "rules.bolt"
```
Comment (anantakrishna) [inline firebase.json]: Shouldn't we add a `firebase-bolt` to the dev dependencies?

Comment (DadImScared) [reply firebase.json]: I think it's preferred to be a global install.

**Pull request merged**